packages/token-bridge-sdk/src/hooks/useArbTokenBridge.ts

```diff
@@ -401,37 +399,45 @@ export const useArbTokenBridge = (
 
   async function depositEthEstimateGas({
     amount,
-    l1Signer
+    from
   }: {
     amount: BigNumber
-    l1Signer: Signer
+    from: string
   }) {
-    const estimatedL1Gas = await gasEstimator.ethDepositL1Gas({
-      amount,
-      l1Signer,
-      l2Provider: l2.provider
+    const depositRequest = await ethBridger.getDepositRequest({ amount, from })
+
+    const estimatedL1Gas = await l1.provider.estimateGas({
+      data: depositRequest.txRequest.data,
+      from: depositRequest.txRequest.from,
+      value: depositRequest.txRequest.value
     })
 
-    const {
-      maxGas: estimatedL2Gas,
-      maxSubmissionCost: estimatedL2SubmissionCost
-    } = await gasEstimator.ethDepositL2Gas(l2.provider)
+    const estimatedL2Gas = await l2.provider.estimateGas({
+      data: depositRequest.txRequest.data,
+      from: depositRequest.txRequest.from,
+      value: depositRequest.txRequest.value
+    })
 
+    // TODO: can we still get estimatedL2SubmissionCost in v3?
+    const estimatedL2SubmissionCost = BigNumber.from(0)
     return { estimatedL1Gas, estimatedL2Gas, estimatedL2SubmissionCost }
   }
 
   async function withdrawEth({
     amount,
     l2Signer,
+    destinationAddress,
     txLifecycle
   }: {
```

Comment:
Let's not change the signature here, let's keep it as before:

```
async function withdrawEth({
  amount,
  l2Signer,
  txLifecycle
}: {
```

We can then pass in `walletAddress` as `destinationAddress` in the implementation:

```
const tx = await ethBridger.withdraw({
  amount,
  l2Signer,
  destinationAddress: walletAddress
})
```